src/utils/ddb/ddb_main.c

```diff
@@ -180,11 +180,35 @@ process_line_cb(void *cb_args, char *line, uint32_t line_len)
 	/* ignore empty lines */
 	if (all_whitespace(line, line_len))
 		return 0;
-	return ddb_run_cmd(ctx, line, ctx->dc_write_mode);
+	return ddb_run_cmd(ctx, line);
 }
 
 #define str_has_value(str) ((str) != NULL && strlen(str) > 0)
 
+static int
+open_if_needed(struct ddb_ctx *ctx, struct program_args *pa, bool *open)
+{
+	int rc = 0;
+
+	if (!str_has_value(pa->pa_pool_path)) {
+		*open = false;
+		goto out;
+	}
+	ctx->dc_pool_path = pa->pa_pool_path;
+
+	if (str_has_value(pa->pa_r_cmd_run)) {
+		rc = ddb_parse_cmd_str(ctx, pa->pa_r_cmd_run, open);
+		if (rc)
+			return rc;
+	}
```

Comment:
What for `else` branch? still need to open?

Reply:
Fixed.